metric_tank/aggmetric.go

```diff
@@ -247,33 +253,117 @@ func (a *AggMetric) addAggregators(ts uint32, val float64) {
 	}
 }
 
-func (a *AggMetric) Persist(c *Chunk) {
-	log.Debug("AggMetric %s Persist(): starting to save %v", a.Key, c)
-	data := c.Series.Bytes()
-	chunkSizeAtSave.Value(int64(len(data)))
+func (a *AggMetric) persist(pos int) {
+	chunk := a.Chunks[pos]
+	chunk.Finish()
+	if *dryRun {
+		chunk.Saved = true
+		return
+	}
 
-	version := FormatStandardGoTsz
-	buf := new(bytes.Buffer)
-	err := binary.Write(buf, binary.LittleEndian, uint8(version))
-	if err != nil {
-		// TODO
+	// this will block when the write queue fills up.
```

Comment:
this statement is misleading. when we fill it up (due to adding a new chunk), the logic continues just fine, unblocked.  the only way this can block is if the channel was already full. the semantics here are debatable but I would personally write something more along the lines of "add chunk to queue, blocking until the queue has the capacity for our chunk"